v1.1.3 release notes (#3022)
pingcap/tidb-operator: CHANGELOG-1.1.md

# TiDB Operator v1.1.3 Release Notes

## Action Required

> - Add a field `cleanPolicy` in `BackupSpec` to denote clean policy for backup data before the object is deleted from the cluster (defaults to `Retain`). If you want to clean backup data as before, please set `spec.cleanPolicy` in `Backup` yaml or `spec.backupTemplate.cleanPolicy` in `BackupSchedule` yaml to `Delete`. ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Add a field `cleanPolicy` in `BackupSpec` to denote the clean policy for backup data when the Backup CR is deleted from the cluster (default to `Retain`). Please note that before v1.1.3, TiDB Operator will clean the backup data in the remote storage when the Backup CR is deleted, so if you want to clean backup data as before, please set `spec.cleanPolicy` in `Backup` CR or `spec.backupTemplate.cleanPolicy` in `BackupSchedule` CR to `Delete`. ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))
```

> - Add a field `cleanPolicy` in `BackupSpec` to denote the clean policy for backup data when the Backup CR is deleted from the cluster (default to `Retain`). Please note that before v1.1.3, TiDB Operator will clean the backup data in the remote storage when the Backup CR is deleted, so if you want to clean backup data as before, please set `spec.cleanPolicy` in `Backup` CR or `spec.backupTemplate.cleanPolicy` in `BackupSchedule` CR to `Delete`. ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Add a field `cleanPolicy` in `BackupSpec` to denote the clean policy for backup data when the Backup CR is deleted from the cluster (default to `Retain`). Note that before v1.1.3, TiDB Operator will clean the backup data in the remote storage when the Backup CR is deleted, so if you want to clean backup data as before, set `spec.cleanPolicy` in `Backup` CR or `spec.backupTemplate.cleanPolicy` in `BackupSchedule` CR to `Delete`. ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))
```

- Replace `mydumper` with `dumpling` for backup.

>   If `spec.mydumper` is configured in the Backup CR or `spec.backupTemplate.mydumper` is configured in the BackupSchedule CR, please migrate it to `spec.dumpling` or `spec.backupTemplate.dumpling` after you upgrade TiDB Operator to v1.1.3, please note that the `spec.mydumper` or `spec.backupTemplate.mydumper` will be lost after the upgrade. ([#2870](https://github.com/pingcap/tidb-operator/pull/2870), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
  If `spec.mydumper` is configured in the Backup CR or `spec.backupTemplate.mydumper` is configured in the BackupSchedule CR, migrate it to `spec.dumpling` or `spec.backupTemplate.dumpling`. After you upgrade TiDB Operator to v1.1.3, note that `spec.mydumper` or `spec.backupTemplate.mydumper` will be lost after the upgrade. ([#2870](https://github.com/pingcap/tidb-operator/pull/2870), [@lichunzhu](https://github.com/lichunzhu))
```

>   If `spec.mydumper` is configured in the Backup CR or `spec.backupTemplate.mydumper` is configured in the BackupSchedule CR, migrate it to `spec.dumpling` or `spec.backupTemplate.dumpling`. After you upgrade TiDB Operator to v1.1.3, note that `spec.mydumper` or `spec.backupTemplate.mydumper` will be lost after the upgrade. ([#2870](https://github.com/pingcap/tidb-operator/pull/2870), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
  If `spec.mydumper` is configured in the `Backup` CR or `spec.backupTemplate.mydumper` is configured in the `BackupSchedule` CR, migrate it to `spec.dumpling` or `spec.backupTemplate.dumpling`. After you upgrade TiDB Operator to v1.1.3, note that `spec.mydumper` or `spec.backupTemplate.mydumper` will be lost after the upgrade. ([#2870](https://github.com/pingcap/tidb-operator/pull/2870), [@lichunzhu](https://github.com/lichunzhu))
```

## Other Notable Changes

- Update tools in backup manager to v4.0.3 ([#3019](https://github.com/pingcap/tidb-operator/pull/3019), [@lichunzhu](https://github.com/lichunzhu))

> - Support cleanPolicy for backup CR to define the clean behavior of the backup data in the remote storage when the backup CR is deleted ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Support `cleanPolicy` for the `Backup` CR to define the clean behavior of the backup data in the remote storage when the `Backup` CR is deleted ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))
```

> - TLS support for TiCDC ([#3011](https://github.com/pingcap/tidb-operator/pull/3011), [@weekface](https://github.com/weekface))

Reviewer 2: ```suggestion
- Add TLS support for TiCDC ([#3011](https://github.com/pingcap/tidb-operator/pull/3011), [@weekface](https://github.com/weekface))
```

> - Add TLS support between drainer and downstream database server ([#2993](https://github.com/pingcap/tidb-operator/pull/2993), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Add TLS support between Drainer and the downstream database server ([#2993](https://github.com/pingcap/tidb-operator/pull/2993), [@lichunzhu](https://github.com/lichunzhu))
```

> - Support specifying mysqlNodePort and statusNodePort for TiDB Service Spec. ([#2941](https://github.com/pingcap/tidb-operator/pull/2941), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Support specifying `mysqlNodePort` and `statusNodePort` for TiDB Service Spec ([#2941](https://github.com/pingcap/tidb-operator/pull/2941), [@lichunzhu](https://github.com/lichunzhu))
```

> - Fix drainer values.yaml `initialCommitTs` bug ([#2857](https://github.com/pingcap/tidb-operator/pull/2857), [@weekface](https://github.com/weekface))

Reviewer 2: ```suggestion
- Fix the `initialCommitTs` bug in Drainer's `values.yaml` ([#2857](https://github.com/pingcap/tidb-operator/pull/2857), [@weekface](https://github.com/weekface))
```

> - Add `backup` config for TiKV server, add `enable-telemetry` and deprecate `disable-telemetry` config for PD server ([#2964](https://github.com/pingcap/tidb-operator/pull/2964), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Add `backup` config for TiKV server, add `enable-telemetry`, and deprecate `disable-telemetry` config for PD server ([#2964](https://github.com/pingcap/tidb-operator/pull/2964), [@lichunzhu](https://github.com/lichunzhu))
```

- Add commitTS info column in `get restore` command ([#2926](https://github.com/pingcap/tidb-operator/pull/2926), [@lichunzhu](https://github.com/lichunzhu))

> - Update used grafana version from v6.0.1 to v6.1.6 ([#2923](https://github.com/pingcap/tidb-operator/pull/2923), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Update the used Grafana version from v6.0.1 to v6.1.6 ([#2923](https://github.com/pingcap/tidb-operator/pull/2923), [@lichunzhu](https://github.com/lichunzhu))
```

> - Support show commitTS in restore status. ([#2899](https://github.com/pingcap/tidb-operator/pull/2899), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Support showing commitTS in restore status ([#2899](https://github.com/pingcap/tidb-operator/pull/2899), [@lichunzhu](https://github.com/lichunzhu))
```

> - Exit without error if the backup data we try to clean does not exist ([#2916](https://github.com/pingcap/tidb-operator/pull/2916), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Exit without error if the backup data the user tries to clean does not exist ([#2916](https://github.com/pingcap/tidb-operator/pull/2916), [@lichunzhu](https://github.com/lichunzhu))
```

> - Support Auto-scaling by Storage for TiKV in TidbClusterAutoScaler ([#2884](https://github.com/pingcap/tidb-operator/pull/2884), [@Yisaer](https://github.com/Yisaer))

Reviewer 2: ```suggestion
- Support Auto-scaling by Storage for TiKV in `TidbClusterAutoScaler` ([#2884](https://github.com/pingcap/tidb-operator/pull/2884), [@Yisaer](https://github.com/Yisaer))
```

> - Support Auto-scaling by Storage for TiKV in `TidbClusterAutoScaler` ([#2884](https://github.com/pingcap/tidb-operator/pull/2884), [@Yisaer](https://github.com/Yisaer))

Reviewer 2: ```suggestion
- Support auto-scaling by storage for TiKV in `TidbClusterAutoScaler` ([#2884](https://github.com/pingcap/tidb-operator/pull/2884), [@Yisaer](https://github.com/Yisaer))
```

- Clean temporary files in `Backup` job with `Dumpling` to save space ([#2897](https://github.com/pingcap/tidb-operator/pull/2897), [@lichunzhu](https://github.com/lichunzhu))

> - Fail the backup job if existing PVC size is smaller than the storage request in the backup job ([#2894](https://github.com/pingcap/tidb-operator/pull/2894), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Fail the backup job if existing PVC's size is smaller than the storage request in the backup job ([#2894](https://github.com/pingcap/tidb-operator/pull/2894), [@lichunzhu](https://github.com/lichunzhu))
```

- Support scaling and auto-failover even if a TiKV store fails in upgrading ([#2886](https://github.com/pingcap/tidb-operator/pull/2886), [@cofyc](https://github.com/cofyc))

> - Fixed a bug where TidbMonitor resource could not be set ([#2878](https://github.com/pingcap/tidb-operator/pull/2878), [@weekface](https://github.com/weekface))

Reviewer 2: ```suggestion
- Fix a bug that the `TidbMonitor` resource could not be set ([#2878](https://github.com/pingcap/tidb-operator/pull/2878), [@weekface](https://github.com/weekface))
```

- Fix an error for the monitor creation in the tidb-cluster chart ([#2869](https://github.com/pingcap/tidb-operator/pull/2869), [@8398a7](https://github.com/8398a7))

> - Remove  `readyToScaleThresholdSeconds` in `TidbClusterAutoScaler`, Operator won't support de-noise in `TidbClusterAutoScaler` ([#2862](https://github.com/pingcap/tidb-operator/pull/2862), [@Yisaer](https://github.com/Yisaer))

Reviewer 2: ```suggestion
- Remove `readyToScaleThresholdSeconds` in `TidbClusterAutoScaler`; TiDB Operator won't support de-noise in `TidbClusterAutoScaler` ([#2862](https://github.com/pingcap/tidb-operator/pull/2862), [@Yisaer](https://github.com/Yisaer))
```

> - Update the lightning used in tidb-backup-manager from v3.0.15 to v4.0.2 ([#2865](https://github.com/pingcap/tidb-operator/pull/2865), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 2: ```suggestion
- Update the version of TiDB Lightning used in tidb-backup-manager from v3.0.15 to v4.0.2 ([#2865](https://github.com/pingcap/tidb-operator/pull/2865), [@lichunzhu](https://github.com/lichunzhu))
```

# TiDB Operator v1.1.3 Release Notes

## Action Required

> - Add a field `cleanData` in `BackupSpec` to denote whether to clean backup data before the object is deleted from the cluster (defaults to false). If you want to clean backup data as before, please set `spec.cleanData` in `Backup` yaml or `spec.backupTemplate.cleanData` in `BackupSchedule` yaml to true. ([#2892](https://github.com/pingcap/tidb-operator/pull/2892), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 3: - Add a field `cleanPolicy` in `BackupSpec` to denote clean policy for backup data before the object is deleted from the cluster (defaults to `Retain`). If you want to clean backup data as before, please set `spec.cleanPolicy` in `Backup` yaml or `spec.backupTemplate.cleanPolicy` in `BackupSchedule` yaml to `Delete`. ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))

Reviewer 3: @cofyc @DanielZhangQD PTAL

- Replace `mydumper` with `dumpling` for backup.
  If `spec.mydumper` is configured in the Backup CR or `spec.backupTemplate.mydumper` is configured in the BackupSchedule CR, please migrate it to `spec.dumpling` or `spec.backupTemplate.dumpling` after you upgrade TiDB Operator to v1.1.3, please note that the `spec.mydumper` or `spec.backupTemplate.mydumper` will be lost after the upgrade. ([#2870](https://github.com/pingcap/tidb-operator/pull/2870), [@lichunzhu](https://github.com/lichunzhu))

## Other Notable Changes

> - update tools in backup manager to v4.0.3 ([#3019](https://github.com/pingcap/tidb-operator/pull/3019), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Update tools in backup manager to v4.0.3 ([#3019](https://github.com/pingcap/tidb-operator/pull/3019), [@lichunzhu](https://github.com/lichunzhu))
```

> - support cleanPolicy for backup CR ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Support cleanPolicy for backup CR to define the clean behavior of the backup data in the remote storage when the backup CR is deleted ([#3002](https://github.com/pingcap/tidb-operator/pull/3002), [@lichunzhu](https://github.com/lichunzhu))
```

- TLS support for TiCDC ([#3011](https://github.com/pingcap/tidb-operator/pull/3011), [@weekface](https://github.com/weekface))
- Add TLS support between drainer and downstream database server ([#2993](https://github.com/pingcap/tidb-operator/pull/2993), [@lichunzhu](https://github.com/lichunzhu))
- Support specifying mysqlNodePort and statusNodePort for TiDB Service Spec. ([#2941](https://github.com/pingcap/tidb-operator/pull/2941), [@lichunzhu](https://github.com/lichunzhu))
- Fix drainer values.yaml `initialCommitTs` bug ([#2857](https://github.com/pingcap/tidb-operator/pull/2857), [@weekface](https://github.com/weekface))

> - add `backup` config for tikv server, add `enable-telemetry` and deprecate `disable-telemetry` config for pd server ([#2964](https://github.com/pingcap/tidb-operator/pull/2964), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Add `backup` config for TiKV server, add `enable-telemetry` and deprecate `disable-telemetry` config for PD server ([#2964](https://github.com/pingcap/tidb-operator/pull/2964), [@lichunzhu](https://github.com/lichunzhu))
```

> - add commitTS info column in `get restore` command ([#2926](https://github.com/pingcap/tidb-operator/pull/2926), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Add commitTS info column in `get restore` command ([#2926](https://github.com/pingcap/tidb-operator/pull/2926), [@lichunzhu](https://github.com/lichunzhu))
```

- Update used grafana version from v6.0.1 to v6.1.6 ([#2923](https://github.com/pingcap/tidb-operator/pull/2923), [@lichunzhu](https://github.com/lichunzhu))
- Support show commitTS in restore status. ([#2899](https://github.com/pingcap/tidb-operator/pull/2899), [@lichunzhu](https://github.com/lichunzhu))

> - Exit without error if we clean not found backup data ([#2916](https://github.com/pingcap/tidb-operator/pull/2916), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Exit without error if the backup data we try to clean does not exist ([#2916](https://github.com/pingcap/tidb-operator/pull/2916), [@lichunzhu](https://github.com/lichunzhu))
```

- Support Auto-scaling by Storage for TiKV in TidbClusterAutoScaler ([#2884](https://github.com/pingcap/tidb-operator/pull/2884), [@Yisaer](https://github.com/Yisaer))

> - Clean temporary files in `Backup` export job to save space ([#2897](https://github.com/pingcap/tidb-operator/pull/2897), [@lichunzhu](https://github.com/lichunzhu))

Reviewer: ```suggestion
- Clean temporary files in `Backup` job with `Dumpling` to save space ([#2897](https://github.com/pingcap/tidb-operator/pull/2897), [@lichunzhu](https://github.com/lichunzhu))
```

- Fail the backup job if existing PVC size is smaller than the storage request in the backup job ([#2894](https://github.com/pingcap/tidb-operator/pull/2894), [@lichunzhu](https://github.com/lichunzhu))
- Support scaling and auto-failover even if a TiKV store fails in upgrading ([#2886](https://github.com/pingcap/tidb-operator/pull/2886), [@cofyc](https://github.com/cofyc))
- Fixed a bug where TidbMonitor resource could not be set ([#2878](https://github.com/pingcap/tidb-operator/pull/2878), [@weekface](https://github.com/weekface))

> - <(.Values.monitor.create) and (.Values.monitor.grafana.create)>: can't give argument to non-function .Values.monitor.create ([#2869](https://github.com/pingcap/tidb-operator/pull/2869), [@8398a7](https://github.com/8398a7))

Reviewer: ```suggestion
- Fix an error for the monitor creation in the tidb-cluster chart ([#2869](https://github.com/pingcap/tidb-operator/pull/2869), [@8398a7](https://github.com/8398a7))
```

- Remove  `readyToScaleThresholdSeconds` in `TidbClusterAutoScaler`, Operator won't support de-noise in `TidbClusterAutoScaler` ([#2862](https://github.com/pingcap/tidb-operator/pull/2862), [@Yisaer](https://github.com/Yisaer))
- Update the lightning used in tidb-backup-manager from v3.0.15 to v4.0.2 ([#2865](https://github.com/pingcap/tidb-operator/pull/2865), [@lichunzhu](https://github.com/lichunzhu))
- Collect metrics for TiCDC, TiDB Lightning and TiKV Importer ([#2835](https://github.com/pingcap/tidb-operator/pull/2835), [@weekface](https://github.com/weekface))

> - Update PD/TiDB/TiKV config to v4.0.2 ([#2828](https://github.com/pingcap/tidb-operator/pull/2828), [@DanielZhangQD](https://github.com/DanielZhangQD))

Reviewer: This is change in v1.1.2.

Reviewer: L31-L35 are all introduced in v1.1.2

Author: these are added by mistake
fixed

> - Collect metrics for TiCDC, TiDB Lightning and TiKV Importer ([#2835](https://github.com/pingcap/tidb-operator/pull/2835), [@weekface](https://github.com/weekface))
> - Update PD/TiDB/TiKV config to v4.0.2 ([#2828](https://github.com/pingcap/tidb-operator/pull/2828), [@DanielZhangQD](https://github.com/DanielZhangQD))
> - Fix the bug that `PD` Member may still be remained after scaling-in ([#2793](https://github.com/pingcap/tidb-operator/pull/2793), [@Yisaer](https://github.com/Yisaer))
> - Support Auto-Scaler Reference in `TidbCluster` Status when there existed `TidbClusterAutoScaler` ([#2791](https://github.com/pingcap/tidb-operator/pull/2791), [@Yisaer](https://github.com/Yisaer))

Author: ```suggestion
```
